store/models.go

```diff
@@ -18,3 +18,20 @@ type Entity struct {
 	Type  string
 	State *homeassistant.State `gorm:"serializer:json"`
 }
+
+// Metric represents a single metric data point
+type Metric struct {
+	ID               uint      `gorm:"primaryKey;autoIncrement"`
+	Timestamp        time.Time `gorm:"index;not null"`
+	MetricType       string    `gorm:"not null;size:50"`
```

Comment:
This is a string but we've defined constants below. Can we make it a custom type?